src/services/reviews.js

```diff
@@ -0,0 +1,672 @@
+
+			//Check if the logged-in user started the review
+			const reviewResource = await reviewResourceQueries.findOne({
+				reviewer_id: userId,
+				resource_id: resourceId,
+			})
+
+			if (!reviewResource || !reviewResource.id) {
+				return await this.createReview(
+					resourceId,
+					reviewType,
+					userId,
+					resource.organization_id,
+					orgId,
+					resource.next_stage,
+					userRoles,
+					resource.type
+				)
+			}
+
+			// Fetch review details if reviewResource exists
+			const review = await reviewsQueries.findOne({
+				organization_id: reviewResource.organization_id,
+				resource_id: resourceId,
+				reviewer_id: userId,
+			})
+
+			if (!review || !review.id) {
+				return responses.failureResponse({
+					message: 'REVIEW_NOT_FOUND',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			//validate resource status
+			const validateResourceStatus = await this.validateResourceStatus(resource.status)
+			if (validateResourceStatus.statusCode !== httpStatusCode.ok) {
+				return validateResourceStatus
+			}
+
+			// Check if the review is already started by someone else
+			if (review.status === common.REVIEW_STATUS_NOT_STARTED) {
+				const reviewCheck = this.validateNoActiveReviewByOthers(resourceId, userId, orgId)
+				if (reviewCheck.statusCode !== httpStatusCode.ok) {
+					return reviewCheck
+				}
+			}
+
+			// Add or update comments
+			if (bodyData.comment) {
+				await handleComments(bodyData.comment, resourceId, userId)
+			}
+
+			//update resource and review table
+			await reviewsQueries.update(
+				{ id: review.id, organization_id: review.organization_id },
+				{ status: common.REVIEW_STATUS_REQUESTED_FOR_CHANGES }
+			)
+
+			await resourceQueries.updateOne(
+				{ organization_id: resource.organization_id, id: resourceId },
+				{ last_reviewed_on: new Date() }
+			)
+
+			return responses.successResponse({
+				statusCode: httpStatusCode.ok,
+				message: 'REVIEW_CHANGES_REQUESTED',
+			})
+		} catch (error) {
+			throw error
+		}
+	}
+
+	/**
+	 * Approve resource
+	 * @method
+	 * @name approveResource
+	 * @param {Object} bodyData - review body data.
+	 * @param {Integer} resourceId - resource id.
+	 * @param {String} userId - logged in user id.
+	 * @param {String} orgId - organization id.
+	 * @returns {JSON} - review approved response.
+	 */
+
+	static async approveResource(resourceId, bodyData, userId, orgId) {
+		try {
+			// Retrieve and validate resource
+			const resource = await resourceQueries.findOne(
+				{
+					id: resourceId,
+				},
+				{ attributes: ['id', 'status', 'organization_id', 'type', 'user_id'] }
+			)
+
+			if (!resource?.id) {
+				return responses.failureResponse({
+					message: 'RESOURCE_NOT_FOUND',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			//validate resource
+			let validateReview = await this.validateReview(resourceId, userId, resource.status, orgId)
```

Comment:
@priyanka-TL Change this variable to ongoingReviw. Add description and sample req/resp